controllers/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -49,13 +53,187 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		}
 		// Error reading the object - requeue the request.
 		logger.Error(err, "unable to get the current NSTemplateTier")
-		return reconcile.Result{}, errs.Wrap(err, "unable to get the current NSTemplateTier")
+		return reconcile.Result{}, fmt.Errorf("unable to get the current NSTemplateTier: %w", err)
 	}
 
 	_, err := toolchainconfig.GetToolchainConfig(r.Client)
 	if err != nil {
-		return reconcile.Result{}, errs.Wrapf(err, "unable to get ToolchainConfig")
+		return reconcile.Result{}, fmt.Errorf("unable to get ToolchainConfig: %w", err)
+	}
+
+	// check if the `status.revisions` field is up-to-date and create a TTR for each TierTemplate
+	if created, err := r.ensureRevision(ctx, tier); err != nil {
+		// todo add/update ready condition false in the NSTemplateTier when something fails
+		return reconcile.Result{}, fmt.Errorf("unable to create new TierTemplateRevision after NSTemplateTier changed: %w", err)
+	} else if created {
+		logger.Info("Requeue after creating a new TTR")
+		return reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
```

Comment:
Is there any reason for such a long delay? As I mentioned in the other comment, one second delay should be ideal in my opinion
```suggestion
		return reconcile.Result{Requeue: true, RequeueAfter: time.Second}, nil
```

Reply:
No particular reason, just wanted to make sure we wait long enough . But we can go have multiple reconciles in case . If that's fine.